code/modules/research/ordnance/doppler_array.dm

```diff
@@ -0,0 +1,288 @@
+/obj/machinery/doppler_array
+	name = "tachyon-doppler array"
+	desc = "A highly precise directional sensor array which measures the release of quants from decaying tachyons. The doppler shifting of the mirror-image formed by these quants can reveal the size, location and temporal affects of energetic disturbances within a large radius ahead of the array.\n"
+	circuit = /obj/item/circuitboard/machine/doppler_array
+	icon = 'icons/obj/machines/research.dmi'
+	icon_state = "tdoppler"
+	base_icon_state = "tdoppler"
+	density = TRUE
+	verb_say = "states coldly"
+	var/cooldown = 10
+	var/next_announce = 0
+	var/max_dist = 150
+	/// Number which will be part of the name of the next record, increased by one for each already created record
+	var/record_number = 1
+	/// List of all explosion records in the form of /datum/data/tachyon_record
+	var/list/records = list()
+	/// Reference to a disk we are going to print to.
+	var/obj/item/computer_hardware/hard_drive/portable/inserted_disk
```

Comment:
What happens if the inserted disk is deleted inside us?

Reply:
Don't think this should ever happen.

Reply:
Will the disk hard delete?

Reply:
Tried to delete an inserted disk but i didn't get any harddel runtimes myself. It in theory should though.

Reply:
could we hold a weakref to the disk rather then a hard one? or clear the reference in Exited if the disk leaves?

Reply:
0825619c26f4b0c6c68b0c68953451c51f0af6cb